src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -1,83 +1,167 @@
+let start;
+let waypoints;
+let end;
+
 document.addEventListener('DOMContentLoaded', function() {
     const elems = document.querySelectorAll('select');
     const instances = M.FormSelect.init(elems, undefined);
 });
 
 
-function initMap() {
-    const directionsRenderer = new google.maps.DirectionsRenderer();
-    const directionsService = new google.maps.DirectionsService();
-    const transitLayer = new google.maps.TransitLayer();
-
-    const map = new google.maps.Map(document.getElementById("map"), {
-        zoom: 10,
-        center: "25 Main Street, Belvedere Tiburon, CA"
-    });
-
-    transitLayer.setMap(map);
-    directionsRenderer.setMap(map);
+function displayMap() {
+    getTripData().then()
 
-    calculateAndDisplayRoute(directionsService, directionsRenderer);
     document.getElementById("mode").addEventListener("change", function() {
-        calculateAndDisplayRoute(directionsService, directionsRenderer);
+        displayRoute(start, waypoints, end);
     });
 
-    document.getElementById("waypoints").addEventListener("change", function() {
-        calculateAndDisplayRoute(directionsService, directionsRenderer);
-    });
 }
 
-
 /**
- * Takes a latitude and longitude pair as parameters and centers the map on that specific
- * Location
+ * Fetches that data from the TripsServlet and stores the hotel PlaceID and the locations PlaceID as variables to use
+ * in the call for geocodePlaceID()
  */
-function setCenter(LatLng) {
-    map.setCenter(LatLng)
-}
 
+async function getTripData(){
+    const response = await fetch("/trip-data");
+    const results = await response.json();
+    console.log(results);
+    const keys = Object.keys(results)
+
+    let hotelID;
+    let locationsArray;
+
+    for (let key of keys){
+        const{hotel} = parseSerializedJson(key);
+        hotelID = hotel;
+        locationsArray = (results[key])
```

Comment:
best to use semicolons at the end of statements in JS, even though it works without them